File: android-app/app/src/main/java/com/lasohealth/android/feature/onboarding/HealthFocus.kt
package com.lasohealth.android.feature.onboarding

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Bedtime
import androidx.compose.material.icons.filled.Bolt
import androidx.compose.material.icons.filled.DirectionsRun
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.FitnessCenter
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector

enum class HealthFocus(
    val displayName: String,
    val icon: ImageVector,
    val color: Color,
) {
    SLEEP("Sleep", Icons.Filled.Bedtime, Color(0xFF5856D6)),
    FITNESS("Fitness", Icons.Filled.DirectionsRun, Color(0xFF4F8A62)),
    HEART_HEALTH("Heart Health", Icons.Filled.Favorite, Color(0xFFCC4B4C)),
    WEIGHT_BODY("Weight & Body", Icons.Filled.FitnessCenter, Color(0xFFD16C37)),
    RECOVERY("Recovery", Icons.Filled.Bolt, Color(0xFFAF52DE)),
}

File: Views/Onboarding/HealthFocus.swift
import SwiftUI

// MARK: - HealthFocus

enum HealthFocus: String, Codable, Identifiable, Hashable, CaseIterable {
    case sleep
    case fitness
    case heartHealth
    case weightBody
    case recovery

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .sleep: "Sleep"
        case .fitness: "Fitness"
        case .heartHealth: "Heart Health"
        case .weightBody: "Weight & Body"
        case .recovery: "Recovery"
        }
    }

    var systemImageName: String {
        switch self {
        case .sleep: "bed.double.fill"
        case .fitness: "figure.run"
        case .heartHealth: "heart.fill"
        case .weightBody: "scalemass.fill"
        case .recovery: "bolt.heart.fill"
        }
    }

    var color: Color {
        switch self {
        case .sleep: .indigo
        case .fitness: .green
        case .heartHealth: .red
        case .weightBody: .orange
        case .recovery: .purple
        }
    }

    /// Maps this focus to one or more HealthCategory values for insight filtering
    var healthCategories: Set<HealthCategory> {
        switch self {
        case .sleep: [.sleep]
        case .fitness: [.activity]
        case .heartHealth: [.heart]
        case .weightBody: [.body]
        case .recovery: [.heart, .respiratory]
        }
    }

    /// All HealthCategory values covered by a set of focuses
    static func categories(for focuses: Set<HealthFocus>) -> Set<HealthCategory> {
        var result = Set<HealthCategory>()
        for focus in focuses {
            result.formUnion(focus.healthCategories)
        }
        return result
    }
}
